File: benchmark-php/app/BenchmarkCsvOutput.php
<?php

namespace Benchmark;


use League\Csv\Writer;


class BenchmarkCsvOutput extends Benchmark
{

	/** @var string */
	public static $fileName = 'php-benchmark-output.csv';

	/** @var string */
	private $outputDir;


	public function __construct(Config $config, $outputDir)
	{
		parent::__construct($config);
		$this->outputDir = $outputDir;
	}


	protected function transformResult($result)
	{
		$rows = parent::transformResult($result);
		for ($i = 0; $i < count($rows); $i++) {
			$rows[$i][0] = 'php - ' . $rows[$i][0];
		}
		return $rows;
	}


	protected function handleResult($headers, $rows)
	{
		$name = $this->outputDir . '/' . self::$fileName;
		$csv = Writer::createFromPath($name, 'w');
		$csv->setDelimiter(';');
		$csv->insertOne($headers);
		$csv->insertAll($rows);
	}
}

File: benchmark-java/src/main/java/benchmark/java/BenchmarkCsvOutput.java
package benchmark.java;

import benchmark.java.metrics.MetricResult;
import com.opencsv.CSVWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;


public class BenchmarkCsvOutput extends Benchmark {

	public static String fileName = "java-benchmark-output.csv";
	private final String outputDir;
	
	
	public BenchmarkCsvOutput(Config config, String outputDir) {
		super(config);
		this.outputDir = outputDir;
	}


	@Override
	protected List<List<String>> transformResult(List<MetricResult> result) {
		
		List<List<String>> rows = super.transformResult(result);
		for (List<String> row : rows) {
			row.set(0, "java - " + row.get(0));
		}
		return rows;
	}
	
	
	
	@Override
	protected void handleResult(List<String> headers, List<List<String>> rows) {
		
		try {
			String name = outputDir + File.separator + fileName;
			CSVWriter writer = new CSVWriter(new FileWriter(name), ';');
			String[] entries = new String[headers.size()];
			headers.toArray(entries);
			writer.writeNext(entries);
			
			for (List<String> row : rows) {
				entries = new String[row.size()];
				row.toArray(entries);
				writer.writeNext(entries);
			}
			writer.close();
		} catch (IOException ex) {
			Logger.getLogger(BenchmarkCsvOutput.class.getName()).log(Level.SEVERE, null, ex);
		}
	}
	
	
}
